### DISCUSSION TITLE
Reordered pe

### DISCUSSION THREAD

**Discussion by wkitlasten** [usgs/pestpp#224] [Q&A]
I'm a bit confused on the idx for oe vs pe, where idx for pe is always 0 to nsubset, but oe seems to retain the original index order. Not sure if/how I messed something up, but I'm not surprised if I did!

Is pe reordered for some reason, but oe is not?
 

> --- running upgrade ensembles --- 
> 
> ...subset idx:pe real name: 0:3, 1:13, 2:23, 3:39...
> 
> ...subset idx:oe real name: 3:3, 13:13, 23:23, 39:39...

version: 5.1.24
binary compiled on Oct  7 2022 at 10:46:45

**Comment by jtwhite79** [comment]
Hey Wes - what you are seeing is that, behind the scenes, if a problem has more than 100,000 pars and you dont say otherwise, ies will dump the lambda-scalefac upgrade ensembles to disk rather than hold them all in memory during testing.  So when this happens, a new par ensemble is formed that just holds the subset realizations.  So that means the index of the par ensemble for testing counts from zero but the realization names  still align up with the observation ensemble.  If you want ies to hold the upgrade ensembles in memory, just pass `ies_upgrades_in_memory true`

**Reply by wkitlasten** [reply]
Makes sense, as always! Thanks.
